object-search/tests/test_objectsearch.py

```diff
@@ -0,0 +1,105 @@
+import base64
+import io
+import json
+import os
+import sys
+import shutil
+import subprocess
+
+import pytest
+
+from jina.flow import Flow
+
+
+os.chdir(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..' ))
+
+
+num_docs = 3
+top_k = 3
+index_flow_file_path = "flow-index.yml"
+query_flow_file_path = "flow-query-original.yml"
+
+
+def config(tmpdir):
+    os.environ["JINA_DATA"] = "tests/test-data/*.jpg"
+    os.environ["JINA_PORT"] = str(45678)
+    os.environ['PARALLEL'] = str(1)
+    os.environ['SHARDS'] = str(1)
+    os.environ['WORKDIR'] = str(tmpdir)
+
+
+def prepare_data():
+    pass
+
+
+def setup_env():
+    subprocess.check_call([sys.executable, "-m", "pip", "install", "-r", "requirements.txt"])
+    subprocess.check_call([sys.executable, "-m", "pip", "install", "requests", "jina[http]"])
+
+setup_env()
```

Comment:
This should be called inside the `test_function` or as a `fixture` (see how in some tests in the core we are setting env variable and deleting with a `generator` trick).

Like this, we would avoid any file importing it to run the `setup_env` when is not needed.

Reply:
I think this is important because I think `pytest` may import all the files in there and then filter the functions starting by `test_`, thus you may conflict pip installs depending on the order of importing and not depending on which example is being run

Reply:
Earlier I kept `setup_env` as test_setup_env() but it felt weird because it wasn't a test. Then I moved it inside the `test_query` but it threw PIL import error due to import of fixture `queries` in the arguments of `test_query`. I like the fixture idea and its working if I make `setup_env` as fixture and pass it as first argument to `test_query`. 

I also moved chdir to this fixture and it has solved the object path problem too :-)

Reply:
Setup env as fixture sounds good